tests/library/browsercontext-cookies-third-party.spec.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ * Modifications copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { contextTest as test, expect } from '../config/browserTest';
+
+test.use({
+  ignoreHTTPSErrors: true,
+});
+
+test(`third party non-partitioned cookies`, async ({ page, browserName, httpsServer }) => {
```

Comment:
I am pretty sure we have quite a few tests for this. Let's consolidate them together I guess?

Reply:
I'm likely missing something, but I looked at all tests that use `Set-Cookie` header and didn't find anything related to third-party cookies. The closes I found was 'iframe should inherit cookies from parent'. Do you have specific tests in mind?

Reply:
Here are some:
- `should be able to send third party cookies via an iframe`
- `should(not) block third party cookies` x3
- `should not block third party SameSite=None cookies` x2

It feels like any tests using `allowsThirdParty` and probably `defaultSameSiteCookieValue` are related to this.

Reply:
Moved all but those in `headful.spec.ts` as I can't put `test.use({ headless: false });` in a describe. Maybe we can call shared functions from there.